sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/WithTimestamps.java

```diff
@@ -101,7 +101,7 @@ public Duration getAllowedTimestampSkew() {
   public PCollection<T> apply(PCollection<T> input) {
     return input
         .apply("AddTimestamps", ParDo.of(new AddTimestampsDoFn<T>(fn, allowedTimestampSkew)))
-        .setTypeDescriptorInternal(input.getTypeDescriptor());
+        .setCoder(input.getCoder());
```

Comment:
I think coder inference propagates input coder if it sees InputT equals
OutputT.

On Tue, Sep 6, 2016, 8:36 AM Thomas Groh <notifications@github.com> wrote:

> In
> sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/WithTimestamps.java
> <https://github.com/apache/incubator-beam/pull/915#discussion_r77659965>:
>
> > @@ -101,7 +101,7 @@ public Duration getAllowedTimestampSkew() {
> >    public PCollection<T> apply(PCollection<T> input) {
> >      return input
> >          .apply("AddTimestamps", ParDo.of(new AddTimestampsDoFn<T>(fn, allowedTimestampSkew)))
> > -        .setTypeDescriptorInternal(input.getTypeDescriptor());
> > +        .setCoder(input.getCoder());
>
> If a coder is explicitly set on the input PCollection (or really any
> upstream PCollection) we should respect that rather than performing coder
> inference again.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-beam/pull/915/files/5aedb8411e24842dc39ad18bc87a3a270ae8b137#r77659965>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACMeH71WwbTRf-QjbYZeGiTDxwUpuOiks5qnYiSgaJpZM4Jz8Wd>
> .
>